cashctrl_ledger/ledger.py

```diff
@@ -47,11 +82,120 @@ def add_price():
         """
         raise NotImplementedError
 
-    def add_vat_code():
+    def add_vat_code(
+        self, code: str, rate: float, account: str, inclusive: bool = True,
+        text: str = ""
+    ):
         """
-        Not implemented yet
+        Adds a new VAT code to the CashCtrl account.
+
+        Parameters:
+            code (str): The VAT code to be added.
+            rate (float): The percentage rate of the VAT, must be between 0 and 1.
+            account (str): The account identifier to which the VAT is applied.
+            inclusive (bool): Determines whether the VAT is calculated as 'NET' 
+                            (True, default) or 'GROSS' (False).
+            text (str): Additional text or description associated with the VAT code.
+
+        Raises:
+            ValueError: If any of the inputs are invalid (wrong type or out of allowed range).
+            Exception: If there is an issue with the server connection or the API request.
         """
-        raise NotImplementedError
+        accounts = self._client.list_accounts()
+        vat_account = accounts.loc[accounts['number'] == account].iloc[0]
+
+        if not vat_account.empty:
+            account = vat_account['id']
+        else:
+            account = None
+
+        payload = {
+            "name": code,
+            "percentage": rate*100,
+            "accountId": account,
+            "calcType": "NET" if inclusive else "GROSS",
+            "documentName": text,
+        }
```

Comment:
Suggestion to improve readability:
- Use a dict to map account 'number's to account 'id's
- TODO: align types between 'account' and account_map.keys().
```
def add_vat_code(...):
    accounts = self._client.list_accounts()
    account_map = accounts.set_index('number')['id'].to_dict()
    payload = {
        "name": code,
        "percentage": rate*100,
        "accountId": account_map.get(account, None),
        "calcType": "NET" if inclusive else "GROSS",
        "documentName": text,
    }
```